src/asciidoctorWebViewConverter.ts

```diff
@@ -202,7 +202,7 @@ export class AsciidoctorWebViewConverter {
       </body>
       </html>`
     }
-    if (nodeName === 'inline_anchor' && node.type === 'link') {
+    if (nodeName === 'inline_anchor' && (node.type === 'link' || node.type === 'xref')) {
```

Comment:
We should implement this logic: https://github.com/asciidoctor/asciidoctor/blob/2983738317abf6bc125c75eede7890c36a10b86c/lib/asciidoctor/converter/html5.rb#L1131-L1154

Reply:
The Ruby code seems quite complex as it resolves the given `refid`s for non-file `xref:`s. Haven't found code yet which would do something similar. Any hints?

Reply:
Asciidoctor.js is transpiled from Ruby to JavaScript. So we should have access to all the Asciidoctor methods.

I can help you implement the non-trivial parts.

Reply:
Currently I have something like this:

```TypeScript
      if (node.type === 'xref') {
        const attrs = []
        if (node.hasAttribute('role')) { attrs.push(` class="${node.getRole()}"`) }

        let text: string
        if (typeof (node.text) === 'string') {
          text = node.text
        }
        const path = node.getAttribute('path')
        if (typeof path === 'string') {
          if (typeof (text) === 'undefined') {
            text = node.getAttribute('path')
          }
        } else {
          // const refid = node.getAttribute('refid')

          // PSEUDO CODE:
          // const refNode = getNodeId(refid)
          //
          // if (refNode.hasAttribute('xrefstyle')) {
          //   const xrefstyle = refNode.getAttribute('xrefstyle')
          //   text = computeText(refnode, xrefstyle)
          // } else {
          //   text = refnode.getTitle()
          // }
        }
        attrs.push(` href="${node.target}"`)
        attrs.push(` data-href="${node.target}"`)
        return `<a${attrs.join()}>${text}</a>`
      }
```

Regular `xref:` to files works quite fine. What I don't know how to solve is the `xref:` to sections in the `PSEUDO CODE` section.
I don't know if there is something like a `computeText` method. I saw that using e.g. `reftext` on sections doesn't change anything in `:toc:`. Don't know if its not handled on purpose or there might be a gap.

Reply:
You should be able to get the catalog `node.document.catalog[:refs]`

```js
node.getDocument().getCatalog()['refs']
```

Reply:
The Opal/Asciidoctor.js syntax for this part `AbstractNode === ref` is probably a bit complex but you can inspect the `ref`, it should contain a class name.

Reply:
@ggrossetie I'd say it looks good:

![grafik](https://github.com/asciidoctor/asciidoctor-vscode/assets/10586564/bcfd296d-8cf4-486a-933f-68686f828046)

wdyt?

Reply:
I'd like to add some tests `src/test/asciidoctorWebViewConverter.test.ts` but am not sure how execute them.

Reply:
One can run tests by
```shell
npm run build && node ./dist/src/test/runTest.js
```

Reply:
@ggrossetie If you could spare some time, could you take a look at the tests?
I added test to `src/test/asciidoctorWebViewConverter.test.ts` but the ones referring to files are failing because expected / actual are not matching.
In the actual output all `.adoc` suffixes are replaced by `.html` which is really not expected.

Reply:
Thanks for taking the time to fix this issue. I will try to find some time tomorrow to push this fix to the finish line.

Reply:
Since we are generating an HTML page, Asciidoctor will use `outfilesuffix` / `relfilesuffix`: 

> Since the default suffix for inter-document cross references in the html5 backend is .html, the resulting link created in these environments may end up pointing to non-existent HTML files. In this case, you need to change the inter-document cross references to refer to other AsciiDoc source files instead.

https://docs.asciidoctor.org/asciidoc/latest/macros/inter-document-xref/#navigating-between-source-files

Reply:
So should I change the test or the backend. As the change is for the preview, I'd assume the backend?

Reply:
The preview should use the `.adoc` suffix since we will navigate between AsciiDoc files.

I don't recall if we are already doing it but we should set `:relfilesuffix: .adoc` in the preview.

You can update the convert function in the test to use `:relfilesuffix: .adoc` as well. Your assertions should be valid.

Reply:
> I don't recall if we are already doing it but we should set :relfilesuffix: .adoc in the preview.

This could be the place where `:relfilesuffix: .adoc` is set for the preview:
https://github.com/asciidoctor/asciidoctor-vscode/blob/acbb89000276bc1541b2c66385009f59645b3d60/src/features/asciidoctorAttributesConfig.ts#L18-L23

Reply:
Yes that's the one 👍🏻